File: annotations-runtime/src/main/java/com/github/dynamicextensionsalfresco/actions/ParameterMapping.java
package com.github.dynamicextensionsalfresco.actions;

import com.github.dynamicextensionsalfresco.actions.annotations.ActionMethod;

import org.alfresco.service.cmr.action.ParameterDefinition;

/**
 * Represents a mapping of an Action parameter to a parameter of an {@link ActionMethod}-annotated method.
 * 
 * @author Laurens Fridael
 * 
 */
class ParameterMapping {

	private final String name;

	private final int index;

	private final Class<?> expectedType;

	private final boolean multivalued;

	private final boolean mandatory;

	ParameterMapping(final ParameterDefinition parameterDefinition, final int index, final Class<?> expectedType) {
		this.name = parameterDefinition.getName();
		this.index = index;
		this.expectedType = expectedType;
		this.multivalued = parameterDefinition.isMultiValued();
		this.mandatory = parameterDefinition.isMandatory();
	}

	public String getName() {
		return name;
	}

	public int getIndex() {
		return index;
	}

	public Class<?> getExpectedType() {
		return expectedType;
	}

	public boolean isMultivalued() {
		return multivalued;
	}

	public boolean isMandatory() {
		return mandatory;
	}

}

File: annotations-runtime/src/main/kotlin/com/github/dynamicextensionsalfresco/actions/ParameterMapping.kt
package com.github.dynamicextensionsalfresco.actions

import org.alfresco.service.cmr.action.ParameterDefinition

/**
 * Represents a mapping of an Action parameter to a parameter of an [ActionMethod]-annotated method.

 * @author Laurens Fridael
 */
class ParameterMapping(parameterDefinition: ParameterDefinition, val index: Int) {

    val name: String

    val isMultivalued: Boolean

    val isMandatory: Boolean

    init {
        this.name = parameterDefinition.name
        this.isMultivalued = parameterDefinition.isMultiValued
        this.isMandatory = parameterDefinition.isMandatory
    }
}
